src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs

```diff
@@ -251,7 +260,27 @@ public void Pack(string nuspecPath, string nupkgPath, Manifest manifest, bool pa
 
                 string baseDirectoryPath = (string.IsNullOrEmpty(BaseDirectory)) ? Path.GetDirectoryName(nuspecPath) : BaseDirectory;
                 builder.Populate(manifest.Metadata);
-                builder.PopulateFiles(baseDirectoryPath, manifest.Files);
+                builder.PopulateFiles(baseDirectoryPath, manifest.Files.Where(m =>
+                {
+                    if (!IgnoreMissingFiles)
+                    {
+                        return true;
+                    }
+
+                    PackageBuilder temp = new PackageBuilder();
+                    bool didAdd = false;
+
+                    try {
+                        temp.AddFiles(baseDirectoryPath, m.Source, m.Target, m.Exclude);
```

Comment:
I see.  This is a lot uglier than I was expecting :/.  As we discussed, maybe have a look at pkgproj conversion and if it isn't too much work that might be better.